### ISSUE TITLE
2020-01-23 tc010 when all concentrations are zero in a profile, parameter results MUST follow the computation engine specification for reporting

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#163]
The computation engine specifications document provides clear details for results of parameters  when concentrations for a profile are all zero.

As of commit 73826ef , the following behavior for tc010 indicates that these data checks/reporting rules are not implemented completely.

The following rules include any normalized values, i.e. AUCLAST/AUCLASTC/AUCLASTDN would all be reported similarly.

### MRTLAST/MRTEVIFO/P must be reported as NA if AUCLAST <= 0 or AUCINFO == NA or AUCTAU == NA

However, in testing with tc010, M1SD, the results indicate that these rules are not completely implemented:

Here we see that 
- AUCLAST
- AUCLASTC
- AUCLASTDN 
- AUC1-12 (AUCT)
- partial AUCs (AUC0_7 etc) 

are reported as NA rather than zero

conversely
- AUCINFO/P 
- AUCINFODN, AUNCINFPDN
- AUMCINFO
- AUMCINFP
- AUCXPCTO/P 
- AUMCXPCTO/P

are reported as zero rather than the required NA

```r
> r
      SDEID DOSE1 DOSEC CMAX FLGACCEPTPREDOSE CMIN CLAST CEST CMAXDN TMAX FLGACCEPTTMAX TMIN TLAST TLAG KEL KELTMLO KELTMHI KELNOPT KELRSQ KELRSQA FLGACCEPTKEL THALF THALFF LASTTIME AUCALL AUCALLDN AUCLAST AUCLASTC AUCLASTDN AUMCLAST AUC1 AUC2 AUC3 AUC4 AUC5 AUC6
1   2925168     3 3e+06    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA       72      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
2  40876220     1 1e+06    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA       72      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
3  63536292     1 1e+06    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA       72      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
4 111299859     1 1e+06    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA       72      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
5 112146587     3 3e+06    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA       72      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
6 123811839   100 1e+08    0                1   NA    NA   NA      0   NA            NA   NA    NA   NA  NA      NA      NA       0     NA      NA            0    NA     NA      120      0        0      NA       NA        NA        0   NA   NA   NA   NA   NA   NA
  AUC7 AUC8 AUC9 AUC10 AUC11 AUC12 AUCINT1 AUCINT2 AUCINT3 AUCINT4 AUCINT5 AUCINT6 AUCINT7 AUCINT8 AUCINT9 AUCINT10 AUCINT11 AUCINT12 AUC0_7 AUC8_48 AUC0_64 AUC0_168 AUCINFO AUCINFP AUCINFOC AUCINFPC AUCINFODN AUCINFPDN AUMCINFO AUMCINFP MRTLAST MRTEVIFO MRTEVIFP
1   NA   NA   NA    NA    NA    NA   0_0.5     0_1   0_1.5     0_2   0_2.5     0_4     0_8    0_12    0_16     0_24     0_48     0_72     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
2   NA   NA   NA    NA    NA    NA   0_0.5     0_1   0_1.5     0_2   0_2.5     0_4     0_8    0_12    0_16     0_24     0_48     0_72     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
3   NA   NA   NA    NA    NA    NA   0_0.5     0_1   0_1.5     0_2   0_2.5     0_4     0_8    0_12    0_16     0_24     0_48     0_72     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
4   NA   NA   NA    NA    NA    NA   0_0.5     0_1   0_1.5     0_2   0_2.5     0_4     0_8    0_12    0_16     0_24     0_48     0_72     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
5   NA   NA   NA    NA    NA    NA   0_0.5     0_1   0_1.5     0_2   0_2.5     0_4     0_8    0_12    0_16     0_24     0_48     0_72     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
6   NA   NA   NA    NA    NA    NA   0_0.5     0_1     0_2     0_4     0_8    0_12    0_16    0_24    0_48     0_72     0_96    0_120     NA      NA      NA       NA       0       0       NA       NA         0         0        0        0      NA       NA       NA
  AUCXPCTO AUCXPCTP AUMCXPTO AUMCXPTP CLFO CLFOW CLFP CLFPW VZFO VZFOW VZFP VZFPW CONC1 CONC2 CONC3 CONC4 CONC5 CONC6 CONC7 CONC8 CONC9 CONC10 CONC11 CONC12 CONC13 CONCTIME1 CONCTIME2 CONCTIME3 CONCTIME4 CONCTIME5 CONCTIME6 CONCTIME7 CONCTIME8 CONCTIME9
1        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       1.5         2       2.5         4         8        12
2        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       1.5         2       2.5         4         8        12
3        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       1.5         2       2.5         4         8        12
4        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       1.5         2       2.5         4         8        12
5        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       1.5         2       2.5         4         8        12
6        0        0        0        0   NA    NA   NA    NA   NA    NA   NA    NA     0     0     0     0     0     0     0     0     0      0      0      0      0         0       0.5         1       2.0         4       8.0        12        16        24
  CONCTIME10 CONCTIME11 CONCTIME12 CONCTIME13 TIMEU AMOUNTU DOSEU VOLUMEU CONCU KELU  CLU     AUCU       AUMCU    AUCNORMU AURCU CONCNORMU RATEU VOLUMEWU     CLWU    STUDY SITEID   SUBJID RAND TREATXT TRTCD PKCOLL PKBDFLD      PKTERM PERIODU PERIOD VISITU VISIT
1         16         24         48         72    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011002    2    3 mg     B  POINT  PLASMA PF-04634817  PERIOD      2    DAY     1
2         16         24         48         72    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011006    6    1 mg     A  POINT  PLASMA PF-04634817  PERIOD      1    DAY     1
3         16         24         48         72    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011003    3    1 mg     A  POINT  PLASMA PF-04634817  PERIOD      1    DAY     1
4         16         24         48         72    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011008    8    1 mg     A  POINT  PLASMA PF-04634817  PERIOD      1    DAY     1
5         16         24         48         72    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011003    3    3 mg     B  POINT  PLASMA PF-04634817  PERIOD      2    DAY     1
6         48         72         96        120    HR      NA    MG       L NG/ML 1/HR L/HR NG.HR/ML NG.HR.HR/ML NG.HR/ML/MG    NA  NG/ML/MG    NA     L/KG ML/HR/KG B1261002   1001 10011011   11  100 mg     E  POINT  PLASMA PF-04634817  PERIOD      2    DAY     1
  PHASE                      ACTTRT ACTTRTS ACTTRTC       DOSEP DOSE DOSPND ROUTE RACES SEX  HT WT HTUNI WTUNI PKPCOM
1    NA   PF-04634817 3 MG SOLUTION      NA       B PF-04634817    3   DONE  Oral BLACK   M 176 93    CM    KG   <NA>
2    NA   PF-04634817 1 MG SOLUTION      NA       A PF-04634817    1   DONE  Oral BLACK   M 183 99    CM    KG   <NA>
3    NA   PF-04634817 1 MG SOLUTION      NA       A PF-04634817    1   DONE  Oral ASIAN   M 175 76    CM    KG   <NA>
4    NA   PF-04634817 1 MG SOLUTION      NA       A PF-04634817    1   DONE  Oral BLACK   M 181 83    CM    KG   <NA>
5    NA   PF-04634817 3 MG SOLUTION      NA       B PF-04634817    3   DONE  Oral ASIAN   M 175 76    CM    KG   <NA>
6    NA PF-04634817 100 MG SOLUTION      NA       E PF-04634817  100   DONE  Oral BLACK   M 174 83    CM    KG   <NA>
```

**Comment by rudraya** [comment]
Currently these parameters need some clarification:

- **AUCT and Partial AUCs**: In the current logic, if TMAX is NA given that AUCMETHOD is "LINLOG" then the AUCT and Partial AUCs will return NA in order to consider the LINLOG method. The check for all concentration values being equal to 0 occurs after this check. Please let me know how to proceed. 

- **AUCINFO**: In the specifications document, AUCINFO is supposed to return 0 if all concetrations are 0, please let me know if want me to return NA instead. This will also impact the value of **AUCINFODN** if all concentrations are 0

- **AUMCINFO**: In the specifications document, AUMCINFO is supposed to return 0 if all concetrations are 0, please let me know if want me to return NA instead.

- **AUMCINFP**: In the specifications document, AUMCINFP is supposed to return 0 if all concetrations are 0, please let me know if want me to return NA instead.

-  **AUCXPCTO/P**: In the specifications document, AUCXPCTO/P is supposed to return 0 if all concetrations are 0, please let me know if want me to return NA instead.

- **AUMCXPCTO/P**: In the specifications document, AUMCXPCTO/P is supposed to return 0 if all concetrations are 0, please let me know if want me to return NA instead.

**Comment by tensfeldt** [comment]
As per the computation engine specifications document:

If all Concentrations for a profile are zero,

* **AUCT and Partial AUCs**: _Return ZERO_

* **AUCINFO/AUCINFOi**: _Return ZERO_. Derived values such as **AUCINFODN** would also be computed and returned as __ZERO_

* **AUMCINFO/AUMCINFOi**: _Return "NA"_ - since cannot compute _KEL_ value which wouldn't be evaluable under these circumstances.

* **AUCINFP/AUCINFPi**: _Return "NA"_

* **AUMCINFP/AUMCINFPi**: _Return "NA"_ - since cannot compute _Cest_ or _KEL_ value which wouldn't be evaluable under these circumstances.

* **AUCXPCTO/P/i**: _Return "NA"_

* **AUMCXPCTO/P/i**: _Return "NA"_

* **AUMCTAU/i**: _Return "NA"_

Any parameter that's dependent on another parameter that has been evaluated as "NA"/missing should also evaluate as "NA"/missing or not-determined.

**Comment by rudraya** [comment]
This issue should be fixed as of ee41560. Please let me know if there are any additional comments.

**Comment by tensfeldt** [comment]
This issue was retested with commit 231f2da successfully with the tc010 with all concentrations set to zero. The results were as expected from the computation engine specifications and this can be closed.
